### PR TITLE
Add CACHE setting to config file (for embedded users)

### PR DESCRIPTION
**Pull request by sbc100**
See https://github.com/emscripten-core/emscripten/pull/11091

### CODE REVIEW

File: emsdk.py [hunk 1]
```diff
@@ -1481,7 +1482,7 @@ def generate_dot_emscripten(active_tools):
 ''' % temp_dir
 
   if embedded:
-    cfg = cfg.replace(emscripten_config_directory, "' + emsdk_path + '")
+    cfg = cfg.replace("'" + emscripten_config_directory, "emsdk_path + '")
```
Comment (kripken) [inline emsdk.py:1485]: is this intentional? if so, what's going on here?

File: emsdk.py [hunk 2]
```diff
@@ -2623,6 +2624,8 @@ def construct_env(tools_to_activate, permanent):
   if to_unix_path(os.environ.get('EM_CONFIG', '')) != to_unix_path(em_config_path):
     env_vars_to_add += [('EM_CONFIG', em_config_path)]
   if emscripten_config_directory == emsdk_path():
+    # Remove this once emscriptensupport EM_CACHE in config file:
```
Comment (kripken) [inline emsdk.py:2627]: I thought it would just support `CACHE`, not `EM_CACHE`? (wasn't the latter the env var name? maybe I'm misremembering)

Comment (sbc100) [reply emsdk.py:2627]: All config settings can be overridden via environment variables.  The pattern is that for a config settings called `FOO` the environment variable is `EM_FOO`.   So depending on which you are talking about they are both true.

File: emsdk.py (hunk 1 above)
Comment (sbc100) [reply emsdk.py:1485]: Sorry this is general cleanup on the config.

It changes pattern to:

```
CACHE = emsdk_path + '/.emscripten_cache'
NODE_JS = emsdk_path + '/node/12.9.1_64bit/bin/node'
LLVM_ROOT = emsdk_path + '/fastcomp/fastcomp/bin'
BINARYEN_ROOT = emsdk_path + '/fastcomp'
EMSCRIPTEN_ROOT = emsdk_path + '/fastcomp/emscripten'
EMSCRIPTEN_NATIVE_OPTIMIZER = emsdk_path + '/fastcomp/bin/optimizer'
```

From the more redundant form:

```
CACHE = '' + emsdk_path + '/.emscripten_cache'
NODE_JS = '' + emsdk_path + '/node/12.9.1_64bit/bin/node'
LLVM_ROOT = '' + emsdk_path + '/fastcomp/fastcomp/bin'
BINARYEN_ROOT = '' + emsdk_path + '/fastcomp'
EMSCRIPTEN_ROOT = '' + emsdk_path + '/fastcomp/emscripten'
EMSCRIPTEN_NATIVE_OPTIMIZER = '' + emsdk_path + '/fastcomp/bin/optimizer'
```

File: emsdk.py (hunk 2 above)
Comment (sbc100) [reply emsdk.py]: Fixed the comment.

Comment (kripken) [reply emsdk.py]: Thanks, right, I forgot about that env var stuff...

**Pull request merged**